m2cgen/interpreters/mixins.py

```diff
@@ -195,3 +195,46 @@ def _get_subroutine_name(self):
 
     def create_code_generator(self):
         raise NotImplementedError
+
+
+class InfixPowExprMixin(BaseToCodeInterpreter):
+    """
+    This mixin is used for languages that provide the exponentiation operation in a form
+    of an infix operator. Such languages inlcude Haskell, F#, R and others.
+
+    The operator used by default is "**", but it can be overriden by setting an appropriate
+    value in `pow_operator`.
+    """
+
+    pow_operator = "**"
+
+    infix_expressions = [*BaseToCodeInterpreter.infix_expressions, ast.PowExpr]
+
+    def interpret_pow_expr(self, expr, left_precedence=None,
+                           right_precedence=None, **kwargs):
+        base_result = self._do_interpret(
+            expr.base_expr, left_precedence=expr.precedence, **kwargs)
+        exp_result = self._do_interpret(
+            expr.exp_expr, right_precedence=expr.precedence, **kwargs)
+        return self._cg.infix_expression(
+            left=base_result, right=exp_result, op=self.pow_operator,
+            wrap=self._wrap_infix_expr(expr, left_precedence, right_precedence))
```

Comment:
Same as above about code readability.
```suggestion
        return self._cg.infix_expression(
            left=base_result,
            right=exp_result,
            op=self.pow_operator,
            wrap=self._wrap_infix_expr(expr, left_precedence, right_precedence))
```

Reply:
Yeah, I feel like we need to keep up with modern best practices and introduce auto-formatting to the project (eg. `black`). Will address this manually for now and will introduce `black` support in future PRs.